src/imt/side.rs

```diff
@@ -0,0 +1,41 @@
+use crate::types::Price;
+
+/// Type markers for Buy and Sell sides
+#[derive(Debug, Clone, Copy)]
+pub struct Buy;
+#[derive(Debug, Clone, Copy)]
+pub struct Sell;
+
+/// IMT side marker
+pub enum Side {
+    Buy,
+    Sell,
+}
+
+/// Trait for defining order comparison logic
+pub trait OrderSide: Clone {
```

Comment:
To ensure that all types implementing OrderSide have consistent behavior, consider adding Copy and possibly Debug as additional trait bounds.

```suggestion
pub trait OrderSide: Clone + Copy + std::fmt::Debug {
```